unit_tests/test_rest_asset_client.py

```diff
@@ -0,0 +1,294 @@
+        websocket_thread_mock = MagicMock()
+        with patch("gsy_e_sdk.clients.rest_asset_client.WebsocketThread",
+                   return_value=websocket_thread_mock) as mocked_class:
+            client = RestAssetClient(asset_uuid=TEST_ASSET_UUID)
+            websocket_uri = f"{client.websockets_domain_name}/" \
+                            f"{client.simulation_id}/{client.asset_uuid}/"
+
+            mocked_class.assert_called_with(websocket_uri,
+                                            client.domain_name,
+                                            client.dispatcher)
+            assert client.websocket_thread is websocket_thread_mock
+            client.websocket_thread.start.assert_called()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_environment_use_functions")
+    def test_start_websocket_connection_thread_pool_executor_instantiated():
+        thread_pool_executor_mock = MagicMock()
+        with patch("gsy_e_sdk.clients.rest_asset_client.ThreadPoolExecutor",
+                   return_value=thread_pool_executor_mock) as mocked_class:
+            client = RestAssetClient(asset_uuid=TEST_ASSET_UUID)
+
+            mocked_class.assert_called_with(max_workers=MAX_WORKER_THREADS)
+            assert client.callback_thread is thread_pool_executor_mock
+
+    @staticmethod
+    def test_endpoint_prefix(client):
+        expected_ret_val = f"{client.domain_name}/external-connection/api/" \
+                           f"{client.simulation_id}/{client.asset_uuid}"
+        assert client.endpoint_prefix == expected_ret_val
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id")
+    def test_register_request_post_call(client):
+        endpoint = f"{client.endpoint_prefix}/register/"
+        data = {"transaction_id": TEST_TRANSACTION_ID}
+
+        with patch("gsy_framework.client_connections.utils.post_request",
+                   return_value=None) as mocked_func:
+            client.register()
+            mocked_func.assert_called_with(endpoint, data, client.jwt_token)
+
+    @staticmethod
+    def test_register_request_not_posted_return_none(client):
+        with patch("gsy_framework.client_connections.utils.post_request",
+                   return_value=False):
+            ret_val = client.register()
+
+            assert ret_val is None
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id")
+    def test_register_request_posted_waits_for_command_response_and_return_expected(client):
+        with patch("gsy_framework.client_connections.utils.post_request",
+                   return_value=True):
+            client.dispatcher = MagicMock()
+            client.dispatcher.wait_for_command_response.return_value = TEST_COMMAND_RESPONSE
+
+            assert client.register() == TEST_COMMAND_RESPONSE
+            client.dispatcher.wait_for_command_response.assert_called_with(
+                "register", TEST_TRANSACTION_ID, timeout=15 * 60)
```

Comment:
Not sure, but we do not want to test the setting of TEST_COMMAND_RESPONSE, so please use the variable. Thanks!
```suggestion
                "register", TEST_TRANSACTION_ID, timeout= TEST_COMMAND_RESPONSE)
```

Reply:
Here I mimicked the arguments with which `dispatcher.wait_for_command_response `method is called. I agree the `15 * 60` is really specific. If you agree, I can change this assertion for just to assert wether it has been called or not, regardless of the arguments.

I'm afraid your suggestion would not work because TEST_COMMAND_RESPONSE is not an argument used when calling that method. You may have thought of `REGISTER_COMMAND_TIMEOUT` though...

Reply:
```suggestion
                "register", TEST_TRANSACTION_ID, timeout=REGISTER_COMMAND_TIMEOUT)
```
It could work like this and the variable being imported from `rest_asset_client.py` as it is defined there...